### PR TITLE
GH-86 make 'path_ant_filter' and 'ant_filter' optional, 

### LINKED ISSUE #86: Include pattern and exclude pattern are both required when using path_ant_filter.
**Describe the bug**
I am doing some testing of path_ant_filter. I am trying to setup a xray that excludes a path. The provider is is requiring both include_patterns and exlude_patterns. When creating the xray watch using the UI both are not required. Pulling the watch configuration using the REST API should that include_patterns is an empty list. 

**Requirements for and issue**
```
resource "xray_watch" "jf_license_watch" {
  name        = "jf-license-watch"
  active      = true
  project_key = "jf"

  watch_resource {
  type       = "repository"
  name       = artifactory_local_maven_repository.jf_mvn_snapshots.key
  
  path_ant_filter {
    exclude_patterns = ["org/apache/**", "org.apache.**"]
    include_patterns = ["**"]
    # include_patterns should be required is exclude_patterns is provided.
    # default value of include_patterns and exclude_patterns could be an empty list []
  }

  assigned_policy {
    name = xray_license_policy.jf_license_policy.name
    type = "license"
  }

  watch_recipients = []
}
```

```
# Validation error when not providing include_patterns
Error: Missing required argument

  on ../modules/xray/watch_license.tf line 14, in resource "xray_watch" "license_watch":
  14:       path_ant_filter {

The argument "include_patterns" is required, but no definition was found.

# Validation error when using an empty list for include_patterns
Error: Not enough list items

  with module.xray.xray_watch.license_watch,
  on ../modules/xray/watch_license.tf line 1, in resource "xray_watch" "license_watch":
   1: resource "xray_watch" "license_watch" {

Attribute requires 1 item minimum, but config has only 0 declared.
```

**Expected behavior**
Create a xray watch using the provider that only requires include_patterns or exlude_patterns if path_ant_filter is used.

### PR DESCRIPTION
**Pull request by danielmkn**
Closes #86 
Users can use only one attribute ('path_ant_filter' or 'ant_filter') if needed. Tests and the documentation updated.

### CODE REVIEW

**Review by alexhung — CHANGES_REQUESTED** [review]
A few typos. I don't see a test for option/empty include or exclude patterns.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,14 @@
-## 1.9.0 (October 18, 2022). Tested on Artifactory 7.46.11 and Xray 3.59.7
+## 1.9.1 (November 17, 2022). Tested on Artifactory 7.46.11 and Xray 3.60.2
+
+BUG FIX:
+
+* resource/xray_watch: removed constraints form 'ant_filter' and 'path_ant_filter' attribute include and exclude fields. 
+Setting both include and exclude filters is not required anymore. The fix allows users to set only one of include/exclude filters, if needed.
+
+Issue [#86](https://github.com/jfrog/terraform-provider-xray/issues/86)
+PR [#]()
```
Comment (alexhung) [inline CHANGELOG.md:9]: Missing PR info.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,14 @@
-## 1.9.0 (October 18, 2022). Tested on Artifactory 7.46.11 and Xray 3.59.7
+## 1.9.1 (November 17, 2022). Tested on Artifactory 7.46.11 and Xray 3.60.2
+
+BUG FIX:
+
+* resource/xray_watch: removed constraints form 'ant_filter' and 'path_ant_filter' attribute include and exclude fields. 
+Setting both include and exclude filters is not required anymore. The fix allows users to set only one of include/exclude filters, if needed.
```
[the remark this answers is not in the document]
Comment (danielmkn) [reply CHANGELOG.md:6]: Rephrased :)

File: CHANGELOG.md (hunk 1 above)
Comment (danielmkn) [reply CHANGELOG.md:9]: Sure thing, added in the next commit.

**Review by alexhung — APPROVED** [review]

**Pull request merged**